File: MoviesAndStuff.Api/MoviesAndStuff.Api/Data/Dtos/Movies/MovieDetailDto.cs
namespace MoviesAndStuff.Api.Data.Dtos.Movies
{
    public class MovieDetailDto
    {
        public long Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string? Review { get; set; }
        public string? Director { get; set; }
        public long? GenreId { get; set; }
        public string? GenreName { get; set; }
        public int? Duration { get; set; }
        public decimal? Rating { get; set; }
        public DateTime? PremiereDate { get; set; }
        public DateTime? WatchDate { get; set; }
        public bool IsWatched { get; set; }
    }
}

File: MoviesAndStuff.Front/src/app/movies/dtos/movie-detail-dto.ts
export interface MovieDetailDto {
  id: number;
  title: string;
  review?: string;
  director?: string;
  genreId?: number;
  genreName?: string;
  duration?: number;
  rating?: number;
  premiereDate?: Date;
  watchDate?: Date;
  isWatched: boolean;
}
